crates/prelude/src/lib.rs

```diff
@@ -104,8 +104,10 @@ macro_rules! applet {
         use wasefire::*;
 
         #[export_name = "main"]
+        #[allow(unreachable_code)]
```

Comment:
Some main functions also return bottoms, so I had to add some of these to suppress warnings. Not sure if this okay.

Reply:
Yes, this is fine. I don't think we have a choice.

Reply:
I also noticed that the trait is now visible from the user applet code. For example, `().maybe_unwrap()` is entirely possible, do you think this would be an issue?

Reply:
Good question. I see a few options:
1. Keep as is.
2. Rename `MaybeUnwrap` and `maybe_unwrap` by prefixing an underscore.
3. Move `MaybeUnwrap` in a new `_internal` module then `use $crate::_internal::MaybeUnwrap as _` when needed.
4. Create a (private) macro to generate the `applet!()` macros such that the `MaybeUnwrap` trait and impls are defined locally to the "applet main" function.

(4) is probably the best but maybe a bit overkill. So (3) seems like the best balance, in particular because (1) and (2) still import the trait in scope.

What do you think?

Reply:
Went with (3), I put the module in another file, do you think just having a `mod _internal {}` block in `prelude/src` would be better?

Reply:
No, that's actually a good idea to have a separate file in case we have to increase its content.